python/lsst/daf/butler/core/ddl.py

```diff
@@ -249,9 +251,14 @@ def process_bind_param(self, value: Any, dialect: sqlalchemy.Dialect) -> Optiona
         else:
             return "%.32x" % value.int
 
-    def process_result_value(self, value: Optional[str], dialect: sqlalchemy.Dialect) -> Optional[uuid.UUID]:
+    def process_result_value(
+        self, value: str | uuid.UUID | None, dialect: sqlalchemy.Dialect
+    ) -> Optional[uuid.UUID]:
         if value is None:
             return value
+        elif isinstance(value, uuid.UUID):
+            # sqlalchemy 2 converts to UUID internally
```

Comment:
Does that mean in sqlalchemy 1.4 we can't do this?

Reply:
In 1.4 strings are returned for UUIDs for both Postgres and sqlite, strings are still handled in `else` clause below.